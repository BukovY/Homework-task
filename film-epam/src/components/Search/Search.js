import React from 'react';
import search from './../../src/img/search.png'
import './Search.css'

const Search = () => {
    return (
        <form>
            <input type="text" placeholder='Movies, person, movie, theaters' className='search'/>
            <img src={search} alt='search' className='search_icon'/>
        </form>
    )
}

export default Search