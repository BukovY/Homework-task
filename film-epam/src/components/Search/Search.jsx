import React from "react";
import searchIcon from "../../static/img/search.png";
import styles from "./Search.module.sass";
import { useSelector, useDispatch } from "react-redux";
import { setSearchValue } from "../../redux/actions/appAction";
import {
  isNeedUpdateSearch,
  setSearchPage,
  setSearchRender,
} from "../../redux/actions/searchAction";

const Search = () => {
  const { search } = useSelector((state) => state.appReducer);
  const dispatch = useDispatch();

  const goToSearch = (e) => {
    e.preventDefault();
    dispatch(setSearchPage(1));
    dispatch(setSearchRender(true));
    dispatch(isNeedUpdateSearch(true));
  };
  return (
    <form>
      <input
        type="text"
        placeholder="Movies, person, movie, theaters"
        className={styles.search}
        value={search}
        onChange={(el) => dispatch(setSearchValue(el.target.value))}
      />
      <button onClick={(e) => goToSearch(e)}>
        <img src={searchIcon} alt="search" className={styles.search_icon} />
      </button>
    </form>
  );
};

export default Search;
