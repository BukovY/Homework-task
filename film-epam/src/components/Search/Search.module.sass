.search
  min-width: 400px
  height: 25px
  color: rgb(212, 209, 209)
  font-size: 14px
  background-color: white
  border-radius: 5px

.search_icon
  position: relative
  width: 20px
  height: 20px
  top: 5px
  right: 30px
  cursor: pointer
