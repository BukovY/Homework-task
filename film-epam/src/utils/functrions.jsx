export const genresIndexToString = (arr, map) => {
  const genresToRender = [];
  for (let i of map) {
    if (arr.indexOf(i.id) !== -1) {
      genresToRender.push(i.name);
    }
  }
  return genresToRender;
};

export const minToTime = (num) => {
  let min = num % 60;
  let hour = (num - min) / 60;
  return String(hour) + ":" + String(min);
};

export const getPhotoCard = (path) => {
  return path === null || path === undefined
    ? "https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
    : `https://image.tmdb.org/t/p/w1280/${path}`;
};

export const getPeopleCard = (img) => {
  return img === null || img === undefined
    ? "https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
    : `https://image.tmdb.org/t/p/w500/${img}`;
};

export const getFilmCover = (img) => {
  return img === null || img === undefined
    ? "https://upload.wikimedia.org/wikipedia/commons/6/65/No-Image-Placeholder.svg"
    : `https://image.tmdb.org/t/p/w500/${img}`;
};

export const getPaginationRange = (maxPage) =>
  Array(maxPage)
    .fill(1)
    .reduce((prev, next, index) => {
      return [...prev, index + 1];
    }, []);

export const getIndexLanguage = (lang) => {
  return lang === "EN" ? 0 : lang === "RU" ? 1 : 2;
};

export const matchOnlyNumber = (str) => /^\d+$/.test(str);
