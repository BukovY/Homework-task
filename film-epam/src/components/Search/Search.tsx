import React, { FC } from "react";
import { useSelector, useDispatch } from "react-redux";
import { setSearchValue } from "../../redux/actions/appAction";
import { setSearchPage } from "../../redux/actions/searchAction";
import { headerTranslation } from "../../static/Translation";
import { getSearchData } from "../../redux/reducers/searchReducers";
import { useHistory } from "react-router-dom";
import SearchIcon from "@material-ui/icons/Search";
import InputBase from "@material-ui/core/InputBase";
import { alpha, makeStyles } from "@material-ui/core/styles";
import Box from "@material-ui/core/Box";
import { RootState } from "../../redux/store";
import { getIndexLanguage } from "../../utils/functrions";

const useStyles = makeStyles((theme) => ({
  search: {
    position: "relative",
    borderRadius: theme.shape.borderRadius,
    backgroundColor: alpha(theme.palette.common.white, 0.15),
    "&:hover": {
      backgroundColor: alpha(theme.palette.common.white, 0.25),
    },
    marginLeft: 0,
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      marginLeft: theme.spacing(1),
      width: "auto",
    },
  },
  searchIcon: {
    padding: theme.spacing(0, 2),
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
  },
  inputRoot: {
    color: "inherit",
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 0),
    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("sm")]: {
      width: "300px",
      "@media screen and (max-width: 670px)": {
        width: "200px",
      },
    },
  },
}));

export const Search: FC = () => {
  const classes = useStyles();
  const { search, languageSelected } = useSelector(
    (state: RootState) => state.app
  );
  const dispatch = useDispatch();
  const history = useHistory();

  const goToSearch = (e: any) => {
    e.preventDefault();
    if (search.length !== 0) {
      dispatch(setSearchPage(1));
      const input = {
        search,
        searchPage: 1,
        languageSelected,
      };
      dispatch(getSearchData(input));
      history.push(`/search/${search.split(" ").join("%20")}`);
    } else {
      alert("Введите поисковой элемент");
    }
  };
  const indLang = getIndexLanguage(languageSelected);

  return (
    <form>
      <Box className={classes.search}>
        <Box className={classes.searchIcon}>
          <SearchIcon />
        </Box>
        <InputBase
          onChange={(el) => dispatch(setSearchValue(el.target.value))}
          value={search}
          placeholder={headerTranslation[indLang]}
          classes={{
            root: classes.inputRoot,
            input: classes.inputInput,
          }}
          inputProps={{ "aria-label": "search" }}
        />
      </Box>
      <button onClick={(e) => goToSearch(e)} style={{ display: "none" }}>
        search
      </button>
    </form>
  );
};
