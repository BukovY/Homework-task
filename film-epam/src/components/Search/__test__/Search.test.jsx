import React from "react";
import rerender from "react-test-renderer";
import { Search } from "../Search";
import { wrapperProvider } from "../../../utils/functrions";

describe("Testing <Search/> Component", () => {
  it("renders correctly", () => {
    const tree = rerender.create(wrapperProvider(<Search />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
