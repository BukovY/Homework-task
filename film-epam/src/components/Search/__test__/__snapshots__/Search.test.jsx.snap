// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Search/> Component renders correctly 1`] = `
<form>
  <input
    className="search"
    onChange={[Function]}
    placeholder="Movies, person, movie, theaters"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    <img
      alt="search"
      className="search_icon"
      src="search.png"
    />
  </button>
</form>
`;
